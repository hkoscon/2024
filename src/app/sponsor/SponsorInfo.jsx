import PropTypes from 'prop-types';

import './sponsorInfo.scss';

export default function SponsorInfo({
  title, logo, description, links,
}) {
  return (
    <div className="columns">
      <div className="column">
        <div className="card">
          <div className="card-content">
            <div className="content">
              <div className="columns sponsorInfo">
                <div className="column is-3">
                  <div className="card-image">
                    <figure className="image sponsorInfo__logo">
                      <img src={logo} alt={title} />
                    </figure>
                  </div>
                </div>
                <div className="column is-9">
                  <h4 className="title">{title}</h4>
                  {/* eslint-disable-next-line react/no-danger */}
                  <div className="content" dangerouslySetInnerHTML={{ __html: description }} />
                  <div className="card-footer">
                    {links.map(({ title: dispalyText, url }) => (
                      <a href={url} key={url} target="_blank" rel="noopener noreferrer" className="card-footer-item">
                        {dispalyText}
                      </a>
                    ))}
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}

SponsorInfo.propTypes = {
  title: PropTypes.string.isRequired,
  logo: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  links: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
  })).isRequired,
};
