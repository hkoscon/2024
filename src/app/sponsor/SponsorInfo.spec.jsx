/* eslint-env jest */

// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react';
import SponsorInfo from './SponsorInfo';

describe('SponsorInfo', () => {
  it('match snapshot', () => {
    const links = [
      { title: 'Test 1', url: 'http://example.com/test1' },
      { title: 'Test 2', url: 'http://example.com/test2' },
    ];
    const { container } = render(
      <SponsorInfo
        title="Testing"
        logo="sponsor.png"
        description="Testing sponsor"
        links={links}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
