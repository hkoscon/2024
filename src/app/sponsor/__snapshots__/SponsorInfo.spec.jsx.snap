// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SponsorInfo match snapshot 1`] = `
<div>
  <div
    class="columns"
  >
    <div
      class="column"
    >
      <div
        class="card"
      >
        <div
          class="card-content"
        >
          <div
            class="content"
          >
            <div
              class="columns sponsorInfo"
            >
              <div
                class="column is-3"
              >
                <div
                  class="card-image"
                >
                  <figure
                    class="image sponsorInfo__logo"
                  >
                    <img
                      alt="Testing"
                      src="sponsor.png"
                    />
                  </figure>
                </div>
              </div>
              <div
                class="column is-9"
              >
                <h4
                  class="title"
                >
                  Testing
                </h4>
                <div
                  class="content"
                >
                  Testing sponsor
                </div>
                <div
                  class="card-footer"
                >
                  <a
                    class="card-footer-item"
                    href="http://example.com/test1"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    Test 1
                  </a>
                  <a
                    class="card-footer-item"
                    href="http://example.com/test2"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    Test 2
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
